internal/pkg/mqtt/sub.go

```diff
@@ -26,27 +34,65 @@ type broker struct {
 	svc    agent.Service
 	client paho.Client
 	logger logger.Logger
+	nats   *nats.Conn
 }
 
 // NewBroker returns new MQTT broker instance.
-func NewBroker(svc agent.Service, client paho.Client, log logger.Logger) MqttBroker {
+func NewBroker(svc agent.Service, client paho.Client, nats *nats.Conn, log logger.Logger) MqttBroker {
 	return &broker{
 		svc:    svc,
 		client: client,
 		logger: log,
+		nats:   nats,
 	}
 }
 
 // Subscribe subscribes to the MQTT message broker
 func (b *broker) Subscribe(topic string) error {
-	s := b.client.Subscribe(topic, 0, b.handleMsg)
+	s := b.client.Subscribe(fmt.Sprintf("%s/%s", topic, reqTopic), 0, b.handleMsg)
 	if err := s.Error(); s.Wait() && err != nil {
 		return err
 	}
 
+	if b.nats != nil {
+		n := b.client.Subscribe(fmt.Sprintf("%s/%s/#", topic, servTopic), 0, b.handleNatsMsg)
+		if err := n.Error(); n.Wait() && err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+// handleNatsMsg triggered when new message is received on MQTT broker
+func (b *broker) handleNatsMsg(mc paho.Client, msg paho.Message) {
+	topic := extractNatsTopic(msg.Topic())
+	b.nats.Publish(topic, msg.Payload())
+}
+
+func extractNatsTopic(topic string) string {
+	i := strings.LastIndex(topic, servTopic) + len(servTopic)
+	natsTopic := topic[i:]
```

Comment:
Validity of the topic can be checked via regexp, there is an example in the CoAP adapter.

This topic handling looks confusing without additional comments of what topic looks like and what you are trying to achieve.